crates/cheatnet/src/execution/cheatable_syscall_handler.rs

```diff
@@ -0,0 +1,297 @@
+    fn execute_hint(
+        &mut self,
+        vm: &mut VirtualMachine,
+        exec_scopes: &mut ExecutionScopes,
+        hint_data: &Box<dyn Any>,
+        constants: &HashMap<String, Felt252>,
+    ) -> HintExecutionResult {
+        let maybe_extended_hint = hint_data.downcast_ref::<Hint>();
+
+        if let Some(Hint::Starknet(StarknetHint::SystemCall { .. })) = maybe_extended_hint {
+            if let Some(Hint::Starknet(hint)) = maybe_extended_hint {
+                return self.execute_next_syscall_cheated(vm, hint);
+            }
+        }
+
+        match contract_print(vm, maybe_extended_hint) {
+            PrintingResult::Printed => return Ok(()),
+            PrintingResult::Passed => (),
+            PrintingResult::Err(err) => return Err(err),
+        }
+        self.syscall_handler
+            .execute_hint(vm, exec_scopes, hint_data, constants)
+    }
+
+    /// Trait function to store hint in the hint processor by string.
+    fn compile_hint(
+        &self,
+        hint_code: &str,
+        ap_tracking_data: &ApTracking,
+        reference_ids: &HashMap<String, usize>,
+        references: &[HintReference],
+    ) -> Result<Box<dyn Any>, VirtualMachineError> {
+        self.syscall_handler
+            .compile_hint(hint_code, ap_tracking_data, reference_ids, references)
+    }
+}
+
+// crates/blockifier/src/execution/syscalls/hint_processor.rs:454
+/// Retrieves a [Relocatable] from the VM given a [`ResOperand`].
+/// A [`ResOperand`] represents a CASM result expression, and is deserialized with the hint.
+fn get_ptr_from_res_operand_unchecked(vm: &mut VirtualMachine, res: &ResOperand) -> Relocatable {
+    let (cell, base_offset) = match res {
+        ResOperand::Deref(cell) => (cell, Felt252::from(0)),
+        ResOperand::BinOp(BinOpOperand {
+            op: Operation::Add,
+            a,
+            b: DerefOrImmediate::Immediate(b),
+        }) => (a, Felt252::from(b.clone().value)),
+        _ => panic!("Illegal argument for a buffer."),
+    };
+    let base = match cell.register {
+        Register::AP => vm.get_ap(),
+        Register::FP => vm.get_fp(),
+    };
+    let cell_reloc = (base + i32::from(cell.offset)).unwrap();
+    (vm.get_relocatable(cell_reloc).unwrap() + &base_offset).unwrap()
+}
+
+pub fn stark_felt_from_ptr_immutable(
+    vm: &VirtualMachine,
+    ptr: &Relocatable,
+) -> Result<StarkFelt, VirtualMachineError> {
+    Ok(felt_to_stark_felt(&felt_from_ptr_immutable(vm, ptr)?))
+}
+
+pub fn felt_from_ptr_immutable(
+    vm: &VirtualMachine,
+    ptr: &Relocatable,
+) -> Result<Felt252, VirtualMachineError> {
+    let felt = vm.get_integer(*ptr)?.into_owned();
+    Ok(felt)
+}
+
+fn get_syscall_operand(hint: &StarknetHint) -> Result<&ResOperand, HintError> {
+    let StarknetHint::SystemCall { system: syscall } = hint else {
+        return Err(HintError::CustomHint(
+            "Test functions are unsupported on starknet.".into(),
```

Comment:
What are "Test functions"? can we improve the error message to be more specific what is not allowed?